mmdet/models/losses/dice_loss.py

```diff
@@ -34,46 +38,15 @@ def dice_loss(pred,
     target = target.flatten(1).float()
 
     a = torch.sum(input * target, 1)
-    b = torch.sum(input * input, 1) + eps
-    c = torch.sum(target * target, 1) + eps
-    d = (2 * a) / (b + c)
-    loss = 1 - d
-    if weight is not None:
-        assert weight.ndim == loss.ndim
-        assert len(weight) == len(pred)
-    loss = weight_reduce_loss(loss, weight, reduction, avg_factor)
-    return loss
-
-
-def naive_dice_loss(pred,
-                    target,
-                    weight=None,
-                    eps=1e-3,
-                    reduction='mean',
-                    avg_factor=None):
-    """Calculate naive dice loss, the coefficient in the denominator is the
-    first power instead of the second power.
-
-    Args:
-        pred (torch.Tensor): The prediction, has a shape (n, *)
-        target (torch.Tensor): The learning label of the prediction,
-            shape (n, *), same shape of pred.
-        weight (torch.Tensor, optional): The weight of loss for each
-            prediction, has a shape (n,). Defaults to None.
-        eps (float): Avoid dividing by zero. Default: 1e-3.
-        reduction (str, optional): The method used to reduce the loss into
-            a scalar. Defaults to 'mean'.
-            Options are "none", "mean" and "sum".
-        avg_factor (int, optional): Average factor that is used to average
-            the loss. Defaults to None.
-    """
-    input = pred.flatten(1)
-    target = target.flatten(1).float()
+    if naive_dice:
+        b = torch.sum(input, 1)
+        c = torch.sum(target, 1)
+        d = (2 * a + eps) / (b + c + eps)
+    else:
+        b = torch.sum(input * input, 1) + eps
+        c = torch.sum(target * target, 1) + eps
+        d = (2 * a) / (b + c)
```

Comment:
Unify the form of `+ eps`, such as:
    b = torch.sum(input if naive_dice else input * input, 1)
    c = torch.sum(target if naive_dice else target * target, 1)
    d = (2 * a) / (b + c + eps)